cmd/prow-job-dispatcher/main.go

```diff
@@ -529,25 +529,59 @@ func clustersMapToSet(clusterMap ClusterMap) sets.Set[string] {
 	return clusterSet
 }
 
-func createPR(o options) {
+func gitCloneRelease() error {
+	cmd := exec.Command("git", "clone", "--depth", "1", "--single-branch", "git@github.com:openshift/release.git")
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to clone repository: %w, output: %s", err, string(output))
+	}
+
+	return nil
+}
+
+func cleanup(directory string) {
+	err := os.RemoveAll(directory)
+	if err != nil {
+		logrus.WithField("directory", directory).WithError(err).Error("failed to remove directory")
+	}
+	logrus.WithField("directory", directory).Info("Successfully removed directory")
+}
+
+func createPR(o options, config *dispatcher.Config) {
+	targetDirWithRelease := filepath.Join(o.targetDir, "/release")
+	cleanup(targetDirWithRelease)
+	defer cleanup(targetDirWithRelease)
+
 	logrus.WithField("targetDir", o.targetDir).Info("Changing working directory ...")
 	if err := os.Chdir(o.targetDir); err != nil {
-		logrus.WithError(err).Fatal("Failed to change to root dir")
+		logrus.WithError(err).Error("failed to change to root dir")
+		return
+	}
+
+	if err := gitCloneRelease(); err != nil {
+		logrus.WithError(err).Error("failed to clone release repository")
+		return
+	}
+
+	if err := dispatcher.SaveConfig(config, filepath.Join(targetDirWithRelease, "/core-services/sanitize-prow-jobs/_config.yaml")); err != nil {
+		logrus.WithError(err).Errorf("failed to save config file to %s", o.configPath)
+		return
 	}
 
 	command := "/usr/bin/sanitize-prow-jobs"
-	arguments := []string{"--prow-jobs-dir", "./ci-operator/jobs", "--config-path", "./core-services/sanitize-prow-jobs/_config.yaml"}
+	arguments := []string{"--prow-jobs-dir", "./release/ci-operator/jobs", "--config-path", "./release/core-services/sanitize-prow-jobs/_config.yaml"}
 	fullCommand := fmt.Sprintf("%s %s", filepath.Base(command), strings.Join(arguments, " "))
 	logrus.WithField("fullCommand", fullCommand).Infof("Running the command ...")
 
 	combinedOutput, err := exec.Command(command, arguments...).CombinedOutput()
 	if err != nil {
-		logrus.WithError(err).WithField("combinedOutput", string(combinedOutput)).Fatal("failed to run the command")
+		logrus.WithError(err).WithField("combinedOutput", string(combinedOutput)).Error("failed to run the command")
```

Comment:
Why switch from Fatal? Is the PR creation still valid when this fails?

Reply:
in this place it should be followed by return, good catch